src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,8 +1,52 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
 public class SalaryInfo {
 
+    private static final DateTimeFormatter DATE_FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder report = new StringBuilder("Report for period ");
+        report.append(dateFrom).append(" - ").append(dateTo).append("\n");
+        List<String> processedNames = new ArrayList<>();
+        Arrays.stream(data)
+                .filter(dataRow -> Arrays.asList(names).contains(dataRow.split(" ")[1]))
+                .forEach(dataRow -> {
+                    String rowName = dataRow.split(" ")[1];
+                    if (!processedNames.contains(rowName)) {
+                        report.append(rowName)
+                                .append(" - ")
+                                .append(getIndividualSalary(data, dateFrom, dateTo, rowName))
+                                .append("\n");
+                        processedNames.add(rowName);
+                    }
+                });
+        return report.deleteCharAt(report.length() - 1).toString();
+    }
+
+    private static int getIndividualSalary(
+            String[] data, String dateFrom, String dateTo, String rowName) {
+        return Arrays.stream(data)
+                .filter(row -> row.split(" ")[1].equals(rowName))
```

Comment:
you repeat some actions several time, for example split one row from "data", so still it's better will be to use loops here